src/standard/BaseStandardVerifier.sol

```diff
@@ -312,87 +367,79 @@ abstract contract BaseStandardVerifier {
+             * We are computing the numerator and denominator and storing them for later for performance gains.
+             * @follow-up The values for σ and σ' need some elaboration. I don't understand where they come from, mostly σ.
+             * @todo Make clearer what this is doing and why such that readers get the reasoning behind it.
              */
             {
-                let gamma := mload(C_GAMMA_LOC)
-                let work_root := mload(OMEGA_LOC)
-                let endpoint := sub(mul(mload(NUM_INPUTS_LOC), 0x20), 0x20)
-                let public_inputs
-                let root_1 := mload(C_BETA_LOC)
-                let root_2 := root_1
+                let beta := mload(C_BETA_LOC) // β
+                let gamma := mload(C_GAMMA_LOC) // γ
+                let work_root := mload(OMEGA_LOC) // ω
                 let numerator_value := 1
                 let denominator_value := 1
 
                 let p_clone := p // move p to the front of the stack
-                let valid := true
+                let valid_inputs := true
 
-                root_1 := mulmod(root_1, 0x05, p_clone) // k1.beta
-                root_2 := mulmod(root_2, 0x07, p_clone) // 0x05 + 0x07 = 0x0c = external coset generator
+                // Load the starting point of the public inputs (jump over the selector and the length of public inputs [0x24])
+                let public_inputs_ptr := add(calldataload(0x24), 0x24)
 
-                public_inputs := add(calldataload(0x24), 0x24)
-                endpoint := add(endpoint, public_inputs)
+                // endpoint_ptr = public_inputs_ptr + num_inputs * 0x20. // every public input is 0x20 bytes
+                let endpoint_ptr := add(public_inputs_ptr, mul(mload(NUM_INPUTS_LOC), 0x20))
 
-                for {} lt(public_inputs, endpoint) {} {
-                    let input0 := calldataload(public_inputs)
-                    let N0 := add(root_1, add(input0, gamma))
-                    let D0 := add(root_2, N0) // 4x overloaded
+                // root_1 = β * 0x05
+                let root_1 := mulmod(beta, 0x05, p_clone) // k1.β
+                // root_2 = β * 0x0c
+                let root_2 := mulmod(beta, 0x0c, p_clone)
+                // @follow-up What is the `root_2` value called?
+                // @note 0x05 + 0x07 == 0x0c == external coset generator
 
-                    root_1 := mulmod(root_1, work_root, p_clone)
-                    root_2 := mulmod(root_2, work_root, p_clone)
+                for {} lt(public_inputs_ptr, endpoint_ptr) { public_inputs_ptr := add(public_inputs_ptr, 0x20) } {
+                    // Load the next public input
+                    let input := calldataload(public_inputs_ptr)
 
-                    let input1 := calldataload(add(public_inputs, 0x20))
-                    let N1 := add(root_1, add(input1, gamma))
+                    // check that wᵢ < p (delays check to the end of the loop for gas savings for success case)
```

Comment:
Yep. The `valid_inputs` is used as a bool that starts as true, and then we perform `&= input < p` for every input. Intention is that we can then just check that `valid_inputs` is true after the loop, if it is, then all the inputs where <p, otherwise we revert the verification.